### ISSUE TITLE
Ability to sort an array of structs by struct field

### ISSUE THREAD

**Issue by david-waterworth** [pola-rs/polars#16110]
### Description

There doesn't appear to be a direct way to sort an array of structs by field, i.e.

Given:

```
df = pl.DataFrame([{"id": 1, "data": [{"key": "A", "value": 2}, {"key": "B", "value": 1}]}])
```

Sort struct by value

```
df = pl.DataFrame([{"id": 1, "data": [{"key": "B", "value": 1}, {"key": "A", "value": 2}]}])
```

There are [work-arounds](https://stackoverflow.com/questions/78440430/sorting-a-polars-liststruct-by-struct-value/78440595#78440595) which revolve around either the side-effect that sorting a struct appears to sort by fields in order so by (re)constructing the struct as {"value": 1, "key": "B"} sorts by value, or by exploding the struct, sorting and then re-grouping.

Ideally `[pl.Expr.list.sort](https://docs.pola.rs/py-polars/html/reference/expressions/api/polars.Expr.list.sort.html)` would accept a `sort key` expression similar to how the python `sorted` function behaves.

If this is accepted I'd be inclined to implement for all sort methods?

**Comment by cmdlineluser** [comment]
Polars has a separate [`Expr.sort_by()`](https://docs.pola.rs/docs/python/dev/reference/expressions/api/polars.Expr.sort_by.html#polars.Expr.sort_by)

So I'm guessing you'd be asking for a list equivalent of that?

```python
df.with_columns(
   pl.col("data").list.sort_by(
       pl.col("data").list.eval(pl.element().struct["value"])
   )
)
# AttributeError: 'ExprListNameSpace' object has no attribute 'sort_by'
```

**Comment by david-waterworth** [comment]
Yeah either works, I guess `list.sort_by` is easier to find (i.e it's explicit), but your 2nd example works for me as well.

**Comment by cr7pt0gr4ph7** [comment]
Are there any general objections to offering a `.list.sort_by(*exprs, **kwargs)` method that is just translated into `.list.eval(pl.element().sort_by(*exprs, **kwargs))` internally (or to just adding an `*exprs` parameter to `.list.sort` that does the same)?

Sounds like a relatively simple addition that improves discoverability of this feature, but I might just be overlooking an important aspect that prevents this.
